internal/services/dns/dns_zone_resource.go

```diff
@@ -374,3 +383,16 @@ func flattenArmDNSZoneSOARecord(input *recordsets.RecordSet) []interface{} {
 
 	return output
 }
+
+func soaRecordHostName(ctx context.Context, timeout time.Duration, recordSetsClient *recordsets.RecordSetsClient, id zones.DnsZoneId) (*string, error) {
```

Comment:
We shall use the `ctx` here directly, instead of create a new `ctx` based on the `timeout`:

```
func soaRecordHostName(ctx context.Context, recordSetsClient *recordsets.RecordSetsClient, id zones.DnsZoneId) (*string, error) {
```

Reply:
ok, fine